service/history/mutableStateBuilder.go

```diff
@@ -455,6 +455,9 @@ func (e *mutableStateBuilder) createNewHistoryEventWithTimestamp(eventID int64,
 	historyEvent.EventId = common.Int64Ptr(eventID)
 	historyEvent.Timestamp = ts
 	historyEvent.EventType = common.EventTypePtr(eventType)
+	if e.replicationState != nil {
```

Comment:
This could be problematic.  I think we update the replicationState at the very end of transaction, so you could write old state version on the history event.

Reply:
ok, i see your point, but should we do the update of the failover version at the very beginning?
history events need to use the failover version, activity / user timer / child workflow, etc, all use that.

Reply:
Sounds good.